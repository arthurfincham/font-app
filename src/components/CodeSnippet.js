import { useState } from 'react';
import Tooltip from '@mui/material/Tooltip';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { grayscale } from 'react-syntax-highlighter/dist/esm/styles/hljs';
import { CopyToClipboard } from 'react-copy-to-clipboard';

export default function CodeSnippet({ language, preview, content }) {
  const [copied, setCopied] = useState(false);

  return (
    <CopyToClipboard text={content}>
      <Tooltip title={copied ? 'Copied!' : language} placement="left">
        <>
          <SyntaxHighlighter language={language} style={grayscale} wrapLongLines={true} onClick={() => setCopied(true)}>
            {preview}
          </SyntaxHighlighter>
        </>
      </Tooltip>
    </CopyToClipboard>
  );
}
