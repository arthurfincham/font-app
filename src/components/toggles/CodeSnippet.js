import ToggleButton from '@mui/material/ToggleButton';
import { StyledToggleButtonGroup } from '../styled/styledTheme';
import ToggleTooltip from '../ToggleTooltip';
import { CodeSnippetContext } from '../../context/CodeSnippetContext';
import { useContext } from 'react';
import CodeIcon from '@mui/icons-material/Code';

export default function Single() {
  const { myCodeSnippet, setMyCodeSnippet } = useContext(CodeSnippetContext);

  const handleToggle = (event, newStyle) => {
    setMyCodeSnippet(newStyle);
  };

  return (
    <ToggleTooltip title="Code Snippet">
      <StyledToggleButtonGroup size="small" value={false} exclusive onChange={handleToggle} aria-label="code snippet">
        <ToggleButton value={!myCodeSnippet} aria-label="left aligned">
          <CodeIcon />
        </ToggleButton>
      </StyledToggleButtonGroup>
    </ToggleTooltip>
  );
}
