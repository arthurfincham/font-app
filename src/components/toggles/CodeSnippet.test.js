import CodeSnippet from './CodeSnippet';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });

it('renders correctly enzyme', () => {
  const wrapper = shallow(<CodeSnippet />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
