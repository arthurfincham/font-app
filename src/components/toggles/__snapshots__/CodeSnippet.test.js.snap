// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<ToggleTooltip
  title="Code Snippet"
>
  <Styled(ForwardRef(ToggleButtonGroup))
    aria-label="code snippet"
    exclusive={true}
    onChange={[Function]}
    size="small"
    value={false}
  >
    <ForwardRef(ToggleButton)
      aria-label="left aligned"
      value={true}
    >
      <Memo(ForwardRef(CodeIcon)) />
    </ForwardRef(ToggleButton)>
  </Styled(ForwardRef(ToggleButtonGroup))>
</ToggleTooltip>
`;
